platforms/core-configuration/groovy-support/src/main/java/org/gradle/api/internal/groovy/support/CompoundAssignmentExtensions.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.groovy.support;
+
+import org.jspecify.annotations.Nullable;
+
+/**
+ * This class provides catch-all support for {@code +=} overloading.
```

Comment:
🤔 Should it be `OP=` here as well, same as for the Result type? This class seems general, not bound to `+=`